File: Zerojudge-h658.cpp
// Author : ysh
// 2024/09/30 Mon 00:35:51
#include<bits/stdc++.h>
using namespace std;
int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);

    int a,b;cin>>a>>b;
    int n;cin>>n;
    vector<pair<int,int>>f(n);
    for(auto &i : f) cin>>i.first>>i.second;

    #define dt(i) (((i.first - a) * (i.first - a)) + ((i.second - b) * (i.second - b)))

    auto ans = *min_element(f.begin(),f.end(), [&](pair<int,int>x,pair<int,int>y) {
        return dt(x) < dt(y);
    });

    cout<<ans.first<<" "<<ans.second;
    return 0;
}

File: Zerojudge-h658.py
# Author : ysh
# 2024/09/30 Mon 00:30:12
a,b = [int(i) for i in input().split()]

f = [[int(i) for i in input().split()] for j in range(int(input()))]
mmin = 10 ** 10
for k in f:
    mmin = min(((a - k[0]) ** 2) + ((b - k[1]) ** 2), mmin)

ans = (([i for i in f if ((a - i[0]) ** 2) + ((b - i[1]) ** 2) == mmin][0]))
print(ans[0], ans[1])

quit()